spec.bs

```diff
@@ -0,0 +1,369 @@
+  <div>
+
+</section>
+
+<section>
+  <h2 id="topics-caller-context-header">Topics caller context</h2>
+  A <dfn>topics-caller-context</dfn> comprises the following fields:
+  1. <dfn>caller-origin</dfn>
+      - For document.browsingTopics() request, the [=caller-origin=] is the document's origin.
+      - For navigation or fetch() request, the [=caller-origin=] is the ongoing request URL's origin (and therefore can change across redirects).
+  1. <dfn>page</dfn>
+      - For document.browsingTopics() request, the [=page=] is the top level page associagted with the document.
+      - For fetch() reqeust initiated from a Window context, the [=page=] is the top level page associated with the fetch() initiator Window.
+      - For navigation request:
+          - If the navigation happens on an iframe:
+            - The [=page=] is the top level page associated with the iframe.
+          - Else (i.e. the navigation happens on the top level browsing context):
+            - The [=page=] is the page that the navigation would result in. Its top level URL should equal the navigation URL.
+
+  Note: a [=page=] can be inactive (e.g. pre-rendered; intermediate redirect for a top navigation), but under all cases it should have a top level URL associated.
+
+  <h2 id="determine-the-topics-of-a-page-header">Determine the topics of a page</h2>
+  The browser should use a model to classify a [=page=] into [=topics=] that are most relevant to the page. The <dfn>input data</dfn> for the model may be the page's domain, URL, text content, etc.
+
+  <h2 id="collect-page-topics-calculation-input-data-header">Collect page topics calculation input data</h2>
+  Given a [=page=] |P|, the <dfn>collect page topics calculation input data</dfn> method steps are:
+  1. Run the following steps in parallel:
+    1. Wait until the necessary [=input data=] |D| for the |P| is ready.
+    1. Wait until |P| has committed and is active.
+    1. Let |E| be the browsing history entry associated with |P|.
+    1. Set |E|.topics-calculation-input-data = |D|.
+
+  <h2 id="collect-topics-caller-origin-header">Collect topics caller origin</h2>
+  Given a [=page=] |P| and a [=caller-origin=] |O|, the <dfn>collect topics caller origin</dfn> method steps are:
+  1. Run the following steps in parallel:
+    1. Wait until |P| has committed and is active.
+    1. Let |E| be the browsing history entry associated with |P|.
+    1. Append |O| to the ordered set |E|.topics-caller-origins.
+
+  <h2 id="periodically-calculate-user-topics-header">Periodically calculate user topics</h2>
+  For a new user agent, the <dfn>user-topics-state</dfn> should be initialized to {epochs: [], hmac-key: a newly generated 128 bit random number}. Then, the [=calculcate user topics=] method should run every 7 days.
+
+  The <dfn>calculcate user topics</dfn> method steps are:
+  1. Let |topics-count| be an empty map.
```

Comment:
Nit: here and throughout, you use a mix of variable declaration styles: `|snake-case|` and `|space separated|`. In modern specs we prefer `|camelCase|`.